lib/at_host/at_host.c

```diff
@@ -194,28 +194,18 @@ static int at_uart_init(char *uart_dev_name)
 static void socket_thread_fn(void *arg1, void *arg2, void *arg3)
 {
 	u8_t at_read_buff[CONFIG_AT_HOST_SOCKET_BUF_SIZE] = {0};
-	int err;
 	int r_bytes;
 
 	ARG_UNUSED(arg1);
 	ARG_UNUSED(arg2);
 	ARG_UNUSED(arg3);
 
 	while (1) {
-		/* Poll the socket for incoming data. */
-		err = poll(fds, nfds, K_FOREVER);
-		if (err < 0) {
-			LOG_ERR("Poll error: %d\n", err);
-		}
-
-		k_mutex_lock(&socket_mutex, K_FOREVER);
```

Comment:
We should get rid of `socket_mutex` entirely then - there's no need to use it for tx-only.

Reply:
The mutexes should now be removed